src/algebra/star/basic.lean

```diff
@@ -117,55 +117,55 @@ export has_trivial_star (star_trivial)
 attribute [simp] star_trivial
 
 /--
-A `*`-semigroup is a semigroup `R` with an involutive operations `star`
+A `*`-magma is a magma `R` with an involutive operations `star`
 so `star (r * s) = star s * star r`.
 -/
-class star_semigroup (R : Type u) [semigroup R] extends has_involutive_star R :=
+class star_magma (R : Type u) [has_mul R] extends has_involutive_star R :=
```

Comment:
We never really use `magma` as a word anywhere. Maybe this is better?
```suggestion
class star_mul (R : Type u) [has_mul R] extends has_involutive_star R :=
```

Reply:
Okay, done.

Reply:
@YaelDillies  Having a field with the same name as the class seems to be a source of confusion:
```
Error: /home/lean/actions-runner/_work/mathlib/mathlib/src/algebra/star/basic.lean:127:0:
invalid simplification lemma 'star_mul'
```
Should I rename the field to something else, like `antihomomorphism`?